### PR TITLE
Pre-commit via make

### PR DESCRIPTION
**Pull request by hiddenmarten**

### CODE REVIEW

**Comment by hiddenmarten** [comment]
As a result, it will look like:
https://github.com/hiddenmarten/etcd-operator/actions/runs/8224339250/job/22488038850?pr=2

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.1
```
Comment (sircthulhu) [inline .github/workflows/pre-commit.yaml:13]: We have version 1.21 in Dockerfile

**Review by sircthulhu — CHANGES_REQUESTED** [review]

File: .github/workflows/pre-commit.yaml (hunk 1 above)
Comment (hiddenmarten) [reply .github/workflows/pre-commit.yaml:13]: Yep, thanks

**Review by sircthulhu — APPROVED** [review]
lgtm

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -73,7 +73,8 @@ var _ = Describe("controller", Ordered, func() {
 
 			By("installing CRDs")
 			cmd = exec.Command("make", "install")
-			_, err = utils.Run(cmd)
+			// TODO: Handle CRD installation error
+			_, _ = utils.Run(cmd)
```
Comment (kvaps) [inline test/e2e/e2e_test.go:77]: Is it still makes sense after https://github.com/aenix-io/etcd-operator/pull/15 get merged?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,6 @@
 # etcd-operator
 New generation community-driven etcd-operator!
+
+## Contributing
+### Prerequisites
+- [pre-commit](https://pre-commit.com/#installation)
```
Comment (kvaps) [inline README.md:6]: Could you rebase PR, please?

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (hiddenmarten) [reply test/e2e/e2e_test.go:77]: Yep, it's a linter validation issue.

File: README.md (hunk 3 above)
Comment (hiddenmarten) [reply README.md]: Sure, done.

**Review by gaalw — APPROVED** [review]

**Review by gecube — APPROVED** [review]

**Pull request merged**